chapters/arrays.tex

```diff
@@ -616,6 +616,106 @@ \subsection{Array Constructor with Iterators}\doublelabel{array-constructor-with
 For deduction of ranges, see \autoref{implicit-iteration-ranges}; and for using types as
 range see \autoref{types-as-iteration-ranges}.
 
+For an iterator:
+\begin{lstlisting}[language=modelica]
+IDENT in class NAME
+\end{lstlisting}
+the \lstinline!NAME! must be (using normal lexical lookup) the
+name of a \lstinline!model!, \lstinline!block!, \lstinline!connector!, \lstinline!record!,
+or \lstinline!operator record!, \lstinline!type! specialized class.
+The loop-variable, \lstinline!IDENT!, is resolved to every instance of class \lstinline!NAME!
+that is on the same or a lower level as the parent of the component containing the iterator expression.
+\begin{nonnormative}
+For example if the iterator expression is in a model A and the simulation contains a component
+of A with component path top.x.a or top.x.a[:] only components in top.x are considered;
+and if the component-path is top.x[n].a only components in top.x[1] are considered for
+top.x[1].a and only top.x[2] for top.x[2].a. If the component path for the model A is a
+all components are considered. The reason for this is to allow graphical composition
+where one sub-component use this and other sub-components are considered.
+\end{nonnormative}
+If no instance of class \lstinline!NAME! is present, the constructed array has zero dimension.
```

Comment:
This is slightly confusing (not as bad as the Rationale document though, which says "zero dimensions"), as it suggests a zero-dimensional array. Perhaps better wording would be, "has size zero".